### PR TITLE
Adjusting some facehugger quirkiness.

### PR DESCRIPTION
**Pull request by Ghommie**
## About The Pull Request
Turns out fast_activate() would call check_lifecycle(), which is a reason facehuggers coming out of trapholes and eggs would die rather quickly. Also fixing infinite facehuggers bug. Larvas being able to nudge eggs regardless.

## Why It's Good For The Game
Tested.

## Changelog
:cl:
add: Alien larvas can now nudge themselves against alien eggs.
fix: Fixing some inconveniences around facehuggers having a shorter lifespan when unleashed by eggs and trapholes.
fix: Fixed a bug with huggers not being properly retrieved by carriers.
/:cl:

### CODE REVIEW

File: code/modules/mob/living/carbon/xenomorph/Castes/Carrier.dm [hunk 1]
```diff
@@ -227,7 +227,10 @@
 /mob/living/carbon/Xenomorph/Carrier/proc/store_hugger(obj/item/clothing/mask/facehugger/F, message = TRUE, forced = FALSE)
 	if(huggers.len < xeno_caste.huggers_max)
 		if(F.stat == CONSCIOUS || forced)
-			temporarilyRemoveItemFromInventory(F)
+			if(F.loc == src)
+				temporarilyRemoveItemFromInventory(F)
+			else
+				F.loc = null
```
Comment (spookydonut) [inline code/modules/mob/living/carbon/xenomorph/Castes/Carrier.dm]: this is bad, dont do this.

File: code/modules/mob/living/carbon/xenomorph/Castes/Carrier.dm [hunk 2]
```diff
@@ -260,7 +263,7 @@
 	var/obj/item/clothing/mask/facehugger/F = get_active_held_item()
 	if(!F) //empty active hand
 		//if no hugger in active hand, we take one from our storage
-		if(huggers.len <= 0)
+		if(length(huggers) <= 0)
```
Comment (spookydonut) [inline code/modules/mob/living/carbon/xenomorph/Castes/Carrier.dm]:
```suggestion
		if(!length(huggers))
```
you cant have negative length

File: code/modules/mob/living/carbon/xenomorph/Castes/Carrier.dm [hunk 3]
```diff
@@ -291,12 +294,9 @@
 		to_chat(src, "<span class='warning'>That egg is tainted!</span>")
 		return
 	if(eggs_cur < xeno_caste.eggs_max)
```
Comment (spookydonut) [inline code/modules/mob/living/carbon/xenomorph/Castes/Carrier.dm]: flip this to an early return.

File: code/modules/mob/living/carbon/xenomorph/Facehuggers.dm [hunk 4]
```diff
@@ -267,8 +266,8 @@
 		fast_activate()
 
 /obj/item/clothing/mask/facehugger/proc/fast_activate(unhybernate = FALSE)
-	if(GoActive(unhybernate))
-		monitor_surrounding()
+	if(GoActive(unhybernate) && loc && !throwing)
```
Comment (Rohesie) [inline code/modules/mob/living/carbon/xenomorph/Facehuggers.dm]: Why the `loc` check?

**Pull request merged**